includes/classes/shopping_cart.php

```diff
@@ -744,173 +767,200 @@ public function calculate()
                     $sale_maker_discount = '';
 
                     // bottom total
-                    if ($attribute_price->fields['product_attribute_is_free'] == '1' && zen_get_products_price_is_free((int)$prid)) {
+                    if ($attribute_price->fields['product_attribute_is_free'] === '1' && zen_get_products_price_is_free($prid)) {
                         // no charge for attribute
+                        continue;
```

Comment:
Brilliant, assuming the bracket-matching indicates this. (This is more a note to myself to trace that back.)

Reply:
Update: Yes. Matched.